rust/src/accumulator.rs

```diff
@@ -0,0 +1,232 @@
+//! This module contains the structure that accumulates usage data
+//! locally before flushing to Kafka.
+//!
+//! The accumulator pre-aggregates usage per timestamp based on
+//! the granularity provided at instantiation.
+//!
+
+use chrono::{DateTime, Duration, DurationRound, Local};
+use std::collections::HashMap;
+use std::fmt;
+use std::mem;
+
+/// The unit of measures we support when recording usage.
+/// more can be added.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum UsageUnit {
+    Milliseconds,
+    Bytes,
+    MillisecondsSec,
```

Comment:
if you want to `derive(Serialize)`, you can use `#[serde(rename = "milliseconds_sec")` for the variant, or `#[serde(rename_all = "snake_case")]` on the whole struct should also work fine.